pyth_observer/events.py

```diff
@@ -555,3 +557,54 @@ def get_event_details(self) -> Tuple[str, List[str]]:
             f"CoinGecko Price Chart: {url}",
         ]
         return title, details
+
+
+class PriceDeviationCrosschain(PriceAccountValidationEvent):
+    """
+    This alert is supposed to fire when a cross-chain price feed deviates
+    from Solana price feed by a specified threshold.
+    """
+
+    error_code: str = "price-deviation-cross-chain"
+    threshold = 20
+
+    def is_valid(self) -> bool:
+        # check if cross-chain price exists
+        if not self.crosschain_price:
+            return True
+
+        trading = (
+            self.price_account.aggregate_price_info.price_status
+            == PythPriceStatus.TRADING
+        )
+        pyth_price = self.price_account.aggregate_price_info.price
+
+        if not trading or pyth_price == 0:
+            return True
+
+        delta = abs(pyth_price - self.crosschain_price["price"])
+        self.crosschain_deviation = delta / self.crosschain_price["conf"]
+
+        # check for stale prices
+        if (
+            self.crosschain_price["publish_time"]
+            == self.crosschain_price["prev_publish_time"]
+        ):
```

Comment:
how could this condition ever be satisfied? this seems like a strange check

Reply:
this was done when I misunderstood that the `publish_time` and `prev_publish_time` were the time the prices were published cross-chain

clarified with @ali-bahjati and it's actually the time it was published on Solana, so have implemented new checking mechanism for stale prices